src/flask_httpauth.py

```diff
@@ -195,6 +195,50 @@ def ensure_sync(self, f):
             return f
 
 
+class HTTPCookieAuth(HTTPAuth):
+    def __init__(self, scheme=None, realm=None, cookie_name=None):
+        super(HTTPCookieAuth, self).__init__(scheme or 'Bearer', realm, cookie_name)
```

Comment:
What is the purpose of scheme or realm when using cookies?

Reply:
Scheme mimics the behavior from `HTTPTokenAuth` with a user-specified `header` so that when `scheme='ApiKey'` the entire cookie value is treated as the token. Any other value of `scheme` is expected to be a space-delimited value like `'<Scheme> <Token>'`. Realm isn't used for authentication, but will be passed along in `WWW-Authenticate` if auth fails.